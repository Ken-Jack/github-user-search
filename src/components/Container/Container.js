import React from 'react';
import {connect} from 'react-fela';
import classNames from 'classnames';

type Props = {
  rootStyle?: any,
  children: any,
  noGutter: boolean,
  styles: Object,
};

function Container({
  noGutter,
  rootStyle,
  children,
  styles,
}: Props) {
  const className = classNames(
    styles.Container_inner,
    {[styles.Container_padding]: noGutter}
  );

  return (
    <div className={classNames(styles.Container_root, rootStyle)}>
      <div className={className}>
        {children}
      </div>
    </div>
  );
}

Container.defaultProps = {
  noGutter: false,
};

const styles = {
  Container_root: () => ({
    margin: 0,
  }),

  Container_inner: () => ({
    margin: '0 auto',
    maxWidth: '1020px',
  }),

  Container_padding: () => ({
    paddingLeft: '10px',
    paddingRight: '10px',

    '@media (min-width: 1050px)': {
      padding: 0,
    },
  }),
};

export default connect(styles)(Container);
