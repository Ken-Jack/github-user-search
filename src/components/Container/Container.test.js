import React from 'react';
import felaSnapshot from 'test-util/fela-snapshot';

import Container from './Container';

describe('Component: Container', () => {

  it('should render', () => {
    const component = felaSnapshot(
      <Container>
        <div>inner component</div>
      </Container>
    );
    expect(component).toMatchSnapshot();
  });

  it('should render style overrides', () => {
    const component = felaSnapshot(
      <Container rootStyle="test">
        <div>inner component</div>
      </Container>
    );
    expect(component).toMatchSnapshot();
  });

});

