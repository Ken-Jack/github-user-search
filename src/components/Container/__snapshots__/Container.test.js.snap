// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Container should render 1`] = `
Object {
  "component": <div
    className="a"
  >
    <div
      className="b c"
    >
      <div>
        inner component
      </div>
    </div>
  </div>,
  "styles": "
    .a{margin:0}
    .b{margin:0 auto}
    .c{max-width:1020}
    .d{padding-left:10}
    .e{padding-right:10}@media (min-width: 1050px){
    .f{padding:0}}",
}
`;

exports[`Component: Container should render style overrides 1`] = `
Object {
  "component": <div
    className="a test"
  >
    <div
      className="b c"
    >
      <div>
        inner component
      </div>
    </div>
  </div>,
  "styles": "
    .a{margin:0}
    .b{margin:0 auto}
    .c{max-width:1020}
    .d{padding-left:10}
    .e{padding-right:10}@media (min-width: 1050px){
    .f{padding:0}}",
}
`;
